gtsam/nonlinear/ISAM2.cpp

```diff
@@ -773,14 +796,22 @@ ISAM2Result ISAM2::update(
     variableIndex_.augment(newFactors, result.newFactorsIndices);
   else
     variableIndex_.augment(newFactors);
+
+  // Augment it with existing factors which now affect to more variables:
+  if (up.newAffectedKeys) {
+    for (const auto &fk : *up.newAffectedKeys) {
+      const auto factorIdx = fk.first;
+      variableIndex_.augmentExistingFactor(factorIdx, fk.second);
```

Comment:
Very scared of this imperative programming construct, but then again, iSAM2 is already scarily imperative. best we can hope for is we have good unit tests and did not forget about scary unforeseen side effects.